tests/console/ansible.pm

```diff
@@ -0,0 +1,181 @@
+    die 'network-engine should be installed!' unless $galaxy_installed =~ m/ansible-network\.network-engine/;
+
+    # 4. Ansible playbook testing
+
+    # Check that community.general.zypper module is available
+    assert_script_run 'ansible-doc -l community.general | grep zypper';
+
+    # Check the playbook
+    assert_script_run 'ansible-playbook -i hosts main.yaml --check';
+
+    # Run the ansible sanity test
+    if (script_run('ansible-test')) {
+        record_soft_failure("boo#1204320 - Ansible: No module named 'ansible_test'");
+    } else {
+        script_run 'ansible-test --help';
+        assert_script_run 'ansible-test sanity';
+    }
+
+    # 5. Ansible playbook execution
+
+    # Print the inventory
+    assert_script_run 'ansible -i hosts all --list-hosts';
+
+    # Run the playbook
+    assert_script_run 'ansible-playbook -i hosts main.yaml';
+
+    # Test that /tmp/ansible/uname.txt created by ansible has desired content
+    my $uname = script_output 'uname -r';
+    validate_script_output 'cat /tmp/ansible/uname.txt', sub { m/$uname/ };
+
+    # Test that /tmp/ansible/static.txt contains the static file
+    validate_script_output 'cat /tmp/ansible/static.txt', sub { m/stay the same/ };
+
+    # Validate that /tmp/ansible/os-release points to /etc/os-release
+    validate_script_output 'readlink /tmp/ansible/os-release', sub { m/\/etc\/os-release/ };
+
+    # Test that /home/johnd/README.txt is readable and contains the expanded template
+    validate_script_output 'sudo -u johnd cat /home/johnd/README.txt', sub { m/my $arch dynamic kingdom/ };
+
+    # Check that Ed - the command line text edit is installed
+    assert_script_run 'which ed';
+
+    # 6. Ansible Vault
+
+    # Generate random password file and random content file
+    my $random_password = random_string(length => 8);
+    my $random_content = random_string(length => 30);
+    assert_script_run "echo '$random_password' > ./ranom_password.txt";
+    assert_script_run "echo '$random_content' > ./ranom_content.txt";
+
+    # Encrypt the content by the password - store it in propperly formated YAML
+    assert_script_run 'echo "---" > ./encrypted_content.yaml';
+    assert_script_run "cat ./ranom_content.txt | ansible-vault encrypt_string --vault-password-file ./ranom_password.txt --stdin-name random_content | tee -a ./encrypted_content.yaml";
+
+    # Decrypt the content and check that it matches the origin
+    validate_script_output 'ansible localhost --vault-password-file ./ranom_password.txt -e "@./encrypted_content.yaml" -m debug -a var=random_content', sub { m/$random_content/ };
+}
+
+sub cleanup {
+    my $self = shift;
+    # Logout $testapi::username and login root
+    enter_cmd 'exit';
+    reset_consoles;
+    $self->select_serial_terminal;
+    assert_script_run 'cd';
+    assert_script_run 'rm -rf ~/ansible_collections/ /tmp/ansible/ /etc/sudoers.d/ansible';
+    zypper_call 'rm ansible git-core python3-yamllint ed';
```

Comment:
I think stopping the Ansible service is needed too, isn't it? other than I'm pretty cool here :D

Reply:
Not valid, there is no systemd unit involved :smile: